src/mlpack/core/data/tokenizers/BERT_tokenizer.hpp

```diff
@@ -0,0 +1,257 @@
+ */
+static std::wstring strip(const std::wstring& text);
+
+/**
+ * This function splits the given text w.r.t. the characters of
+ * stripChar.
+ *
+ * @param text The string which is to be split.
+ */
+static std::vector<std::wstring> split(const std::wstring& text);
+
+/**
+ * Runs basic whitespace cleaning and splitting on a piece of text.
+ *
+ * @param text The string which is stripped and split on.
+ */
+static std::vector<std::wstring> whitespaceTokenize(const std::wstring& text);
+
+/**
+ * Convert a std::string type variable to a std::wstring type variable.
+ *
+ * @param s The input string.
+ * @param ws The output wstring which is the conversion of the input string.
+ */
+void convertToUnicode(const std::string &s, std::wstring &ws);
+
+/**
+ * Convert a std::wstring type variable to a std::string type variable.
+ *
+ * @param ws The input string.
+ * @param s The output wstring which is the conversion of the input wstring.
+ */
+void convertFromUnicode(const std::wstring &ws, std::string &s);
+
+/**
+ * Function for loading the vocab.txt file from the BERT model into a
+ * std::shared_ptr<std::unordered_map<std::wstring, size_t>>.
+ *
+ * @param vocabfile Path of BERT vocab file.
+ */
+static std::shared_ptr<Vocab> loadVocab(const std::string& vocabFile);
+
+/**
+ * Function to replace substring in a given string.
+ *
+ * @param str The string which is to be processed.
+ * @param old_substring The substring which shall be removed.
+ * @param new_substring The substring which is replaced in place of old_substring.
+ */
+std::wstring str_replacer(std::wstring str, std::wstring old_substring, std::wstring new_substring);
+
+/**
+ * The following class is used to run WordPiece tokenization which is a
+ * subword segmentation algorithm used in natural language processing.
+ */
+class WordpieceTokenizer {
+ public:
+  /**
+   * Initialize the WordpieceTokenizer class.
+   *
+   * @param vocab The pointer containing the BERT tokens along with corresponding id's.
+   * @param unkToken Token "[UNK]" of BERT Tokenizer.
+   * @param maxInputCharsPerWord Maximum characters allowed per word.
+   */
+  WordpieceTokenizer(std::shared_ptr<Vocab> vocab, const std::wstring& unkToken = L"[UNK]", size_t maxInputCharsPerWord=200);
+
+  /**
+   * Tokenizes a piece of text into its word pieces.
+   * This uses a greedy longest-match-first algorithm to perform tokenization
+   * using the given vocabulary.
+   * For example:
+   * input = "unaffable"
+   * output = ["un", "##aff", "##able"]
+   *
+   * @param text The input string which is already been passed
+   * through `BasicTokenizer.
+   */
+  std::vector<std::wstring> tokenize(const std::wstring& text) const;
+
+ private:
+  std::shared_ptr<Vocab> mVocab;
+  std::wstring mUnkToken;
+  size_t mMaxInputCharsPerWord;
+}; // class WordpieceTokenizer
+
+/**
+ * This class uses functions and classes previously defined
+ * to run end-to-end tokenziation.
```

Comment:
If this is the user-facing class, please elaborate in the documentation to provide usage examples, etc.  Assume that the user reading this documentation knows nothing about the other classes (and doesn't care to read through the implementation to figure out what it does).